clang/lib/StaticAnalyzer/Checkers/WebKit/RefCntblBaseVirtualDtorChecker.cpp

```diff
@@ -11,16 +11,134 @@
 #include "PtrTypesSemantics.h"
 #include "clang/AST/CXXInheritance.h"
 #include "clang/AST/RecursiveASTVisitor.h"
+#include "clang/AST/StmtVisitor.h"
 #include "clang/StaticAnalyzer/Checkers/BuiltinCheckerRegistration.h"
 #include "clang/StaticAnalyzer/Core/BugReporter/BugReporter.h"
 #include "clang/StaticAnalyzer/Core/BugReporter/BugType.h"
 #include "clang/StaticAnalyzer/Core/Checker.h"
+#include "llvm/ADT/DenseSet.h"
+#include "llvm/ADT/SetVector.h"
 #include <optional>
 
 using namespace clang;
 using namespace ento;
 
 namespace {
+
+class DerefAnalysisVisitor
```

Comment:
🚲🏠: by "deref" you mean "decrement reference count" right? Because when I see it in the compiler I mostly think of operations `*` and `->` and `[]` and such. This is readable when it's next to a similarly named thing named "ref" but in this case I was confused for a few minutes, like what does this have to do with templates?

Reply:
Oh, this is `deref` function (i.e. function by the name of `deref`) analysis visitor.
Maybe I should call it `DerefFuncVisitor` instead?

Reply:
I'd even go with like `UnsafeDeleteVisitor` or something. Because fundamentally you're looking for `delete`, and `deref` is just one of the forms it takes, or something 😅

Reply:
Hm... but it's important that the function in question is `deref()` though. In that, that's what guarantees the object to be eventually deleted appropriately. Maybe `DerefFuncDeleteExprVisitor`?

Reply:
Sounds great to me!